### PR TITLE
Add missing transactions functions

### PR DESCRIPTION
**Pull request by pablocampogo**
- Add `GetBlockTransactions` function
- Update Transactions output to new core version
- Separate `Transaction` structure for easier use on indexer

### CODE REVIEW

File: provider/transaction.go [hunk 1]
```diff
@@ -24,3 +24,78 @@ type SendTransactionInput struct {
...
+				ToAddress   string `json:"to_address"`
+			} `json:"value"`
+		} `json:"msg"`
+		Signature struct {
+			PubKey    string `json:"pub_key"`
+			Signature string `json:"signature"`
+		} `json:"signature"`
+	} `json:"stdTx"`
+	Tx       string `json:"tx"`
+	TxResult struct {
```
Comment (rem1niscence) [inline provider/transaction.go:78]: TxResult varies depending on the type of the transaction, this should be thread in a more general sense

Comment (pablocampogo) [reply provider/transaction.go:78]: Do you propose using `interface{}` or can we be more specific?

Comment (crisog) [reply provider/transaction.go:78]: > TxResult varies depending on the type of the transaction, this should be thread in a more general sense

StdTx is what changes depending on the type of the transaction, TxResult is always the same

Comment (crisog) [reply provider/transaction.go:78]: So, let's give a general structure to that one instead if possible

Comment (crisog) [reply provider/transaction.go:78]: The `Msg` is the only thing that varies btw

Comment (rem1niscence) [reply provider/transaction.go:78]: > The `Msg` is the only thing that varies btw

you're right, on that case `any` can be added to that in favour of `interface{}` as both mean the same and is more modern term

Comment (pablocampogo) [reply provider/transaction.go:74]: Used `interface{}` because that's what I'm using everywhere on the SDK

**Review by rem1niscence — APPROVED** [review]

**Review by crisog — APPROVED** [review]

**Pull request merged**